dvc/data/stage.py

```diff
@@ -336,6 +364,8 @@ def stage(
         logger.debug("staged tree '%s'", obj)
         if name != "md5":
             obj = _stage_external_tree_info(odb, obj, name)
+
+        odb.add(obj.fs_path, obj.fs, obj.hash_info.as_raw())
```

Comment:
Let's add some note here that this is for that tricky optimization or something.